app/uk/gov/hmrc/excisemovementcontrolsystemapi/repository/MovementRepository.scala

```diff
@@ -283,6 +285,33 @@ class MovementRepository @Inject() (
       .toFuture()
       .as(Done)
   }
+
+  def getProblemMovements(): Future[Seq[ProblemMovement]] =
+    collection
+      .aggregate[ProblemMovement](
+        Seq(
+          Aggregates.project(Json.obj("messages.encodedMessage" -> 0).toDocument()),
+          Aggregates.unwind("$messages"),
+          Aggregates.`match`(Json.obj("messages.messageType" -> "IE801").toDocument()),
+          Aggregates.group("$_id", Accumulators.sum("countOfIe801s", 1)),
+          Aggregates.`match`(Filters.gt("countOfIe801s", 2))
```

Comment:
I might misunderstand the problem but won't this only return movements with 3+ 801s? I thought there should only be one?